### PR TITLE
App 1104 remove our erc 1271 sol and import from oz

### PR DESCRIPTION
**Pull request by mathewmeconry**
## Description

Removes custom ERC1271 interface and replaces it with OZ.
Adds test for calls to ERC1271 address.

Task: APP-1104

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [x] I have updated the ``CHANGELOG.md`` file in the root folder.
- [ ] I have tested my code on the test network.

### CODE REVIEW

File: packages/contracts/test/core/dao.ts [hunk 1]
```diff
@@ -398,4 +401,54 @@ describe('DAO', function () {
       expect(await ethers.provider.getBalance(dao.address)).to.equal(amount);
     });
   });
+
+  describe.only('ERC1271', async () => {
```
Comment (novaknole) [inline packages/contracts/test/core/dao.ts:405]: we should remove `only` i think...

Comment (mathewmeconry) [reply packages/contracts/test/core/dao.ts:405]: oohh crap good point. forgot that one

**Review by novaknole — APPROVED** [review]

File: packages/contracts/contracts/core/DAO.sol [hunk 2]
```diff
@@ -207,14 +207,14 @@ contract DAO is
         override
         auth(address(this), SET_SIGNATURE_VALIDATOR_PERMISSION_ID)
     {
-        signatureValidator = ERC1271(_signatureValidator);
+        signatureValidator = IERC1271(_signatureValidator);
```
Comment (Rekard0) [inline packages/contracts/contracts/core/DAO.sol:210]: looking at the code, 
i think we are missing some parts:
- we need to setSignatureValidator() in the init func.
- we need an event related to setSignatureValidator.

File: packages/contracts/test/test-utils/interfaces.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+import { ethers } from 'hardhat';
+
+export function getInterfaceID(contractInterface: ethersDirect.utils.Interface) {
+    let interfaceID = ethers.constants.Zero;
+    const functions: string[] = Object.keys(contractInterface.functions);
+    for (let i = 0; i < functions.length; i++) {
+        interfaceID = interfaceID.xor(contractInterface.getSighash(functions[i]));
+    }
+    return interfaceID.toHexString();
+}
```
Comment (Rekard0) [inline packages/contracts/test/test-utils/interfaces.ts:11]: new line at the end of file.

File: packages/contracts/contracts/test/ERC1271Mock.sol [hunk 4]
```diff
@@ -0,0 +1,9 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.10;
+
+contract ERC1271Mock {
+    function isValidSignature(bytes32, bytes memory) public pure returns (bytes4) {
+        return 0x41424344;
```
Comment (Rekard0) [inline packages/contracts/contracts/test/ERC1271Mock.sol:7]: should we do an actual `ECDSA.recover()` for the deployer address of this mock? 
so we can have more tests?
WDYT?

File: packages/contracts/test/core/dao.ts [hunk 5]
```diff
@@ -398,4 +373,54 @@ describe('DAO', function () {
...
+      ).to.be.eq(true);
+    });
+
+    it('should return 0 if no validator is set', async () => {
+      expect(
+        await dao.isValidSignature(ethers.utils.keccak256('0x00'), '0x00')
+      ).to.be.eq('0x00000000');
+    });
+
+    it('should allow only SET_SIGNATURE_VALIDATOR_PERMISSION_ID to set validator', async () => {
```
Comment (Michael-A-Heuer) [inline packages/contracts/test/core/dao.ts:392]:
```suggestion
    it('should revert if the caller does not have the `SET_SIGNATURE_VALIDATOR_PERMISSION_ID` permissions', async () => {
```

File: packages/contracts/test/test-utils/interfaces.ts (hunk 3 above)
Comment (Michael-A-Heuer) [reply packages/contracts/test/test-utils/interfaces.ts:11]: Please use prettier as the default formatter.
In visual studio code, press `cmd` + `shift` + `p`, then type `Format Document With ...`, then chose `Configure Default Formatter`, and then `Prettier - Code Formatter`.
This requires the [Prettier - Code formatter](https://marketplace.visualstudio.com/items?itemName=esbenp.prettier-vscode) extension

File: packages/contracts/contracts/core/DAO.sol (hunk 2 above)
Comment (Michael-A-Heuer) [reply packages/contracts/contracts/core/DAO.sol:210]: Agreed

**Review by novaknole — APPROVED** [review]

**Review by Rekard0 — APPROVED** [review]

**Pull request merged**